### PR TITLE
Update arborx

### PR DESCRIPTION
**Pull request by Rombur**
Update to use the current master of arborx and add a couple fixes that are necessary to build DTK

### CODE REVIEW

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 1]
```diff
@@ -96,8 +98,9 @@ void sendDataAcrossNetwork( ArborX::Details::Distributor const &distributor,
 }
 
 template <typename T, typename... Targs>
-void sendDataAcrossNetwork( ArborX::Details::Distributor const &distributor,
-                            std::pair<T, T> d, Targs... data )
+void sendDataAcrossNetwork(
+    ArborX::Details::Distributor<typename T::device_type> const &distributor,
+    std::pair<T, T> d, Targs... data )
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: Did you mean to forward the pack?

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 2]
```diff
@@ -387,14 +391,16 @@ PointSearch<DeviceType>::performDistributedSearch(
...
                           Kokkos::RangePolicy<ExecutionSpace>( 0, n_points ),
                           KOKKOS_LAMBDA( int i ) {
-                              queries( i ) = ArborX::within(
-                                  {{points_coord( i, 0 ), points_coord( i, 1 ),
-                                    points_coord( i, 2 )}},
-                                  0. );
+                              queries( i ) = ArborX::intersects( ArborX::Sphere{
+                                  {points_coord( i, 0 ), points_coord( i, 1 ),
+                                   points_coord( i, 2 )},
+                                  0.} );
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: Intersect with a point would be better.

**Comment by Rombur** [comment]
Both are valid comment and they are actually fixed in my branch that uses the `callback` mechanic. This PR just update ArborX, I would prefer that we wait for the `callback` PR to fix these comments.

**Comment by Rombur** [comment]
retest this please

**Comment by Rombur** [comment]
I think this is ready to be merged. The fixes Damien wants will be done in a later PR where we use the new callback mechanism.

**Pull request merged**